### PR TITLE
Publish an event in the Agen format

### PR DESCRIPTION
**Pull request by zrdgthr**
Publish an event in the Agen format when invoking REMREM publish service and cli endpoints

### CODE REVIEW

File: publish-service/src/test/resources/EiffelActivityFinishedEvent.json [hunk 1]
```diff
@@ -1,37 +1,57 @@
...
+    "time": 1554385219068,
+    "tags": [
+      "tag1",
+      "tag2"
+    ],
+    "source": {
+      "domainId": "domainID",
+      "host": "host",
+      "name": "name",
+      "serializer": "pkg:maven",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/EiffelActivityFinishedEvent.json]: Use proper serializer values.

File: publish-service/src/test/resources/EiffelActivityFinishedEvent.json [hunk 2]
```diff
@@ -1,37 +1,57 @@
...
+      "uri": "http://java.sun.com/j2se/1.3/"
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/EiffelActivityFinishedEvent.json]: Use a valid URI. like http://localhost:8080/fem/

File: publish-service/src/test/resources/EiffelActivityFinishedEvent.json [hunk 3]
```diff
@@ -1,37 +1,57 @@
...
+    },
+    "security": {
+      "authorIdentity": "abc",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/EiffelActivityFinishedEvent.json]: nor a valid name.

File: publish-service/src/test/resources/EiffelActivityFinishedEvent.json [hunk 4]
```diff
@@ -1,37 +1,57 @@
...
+      "sequenceProtection": [
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/EiffelActivityFinishedEvent.json]: if not used remove it.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityCanceledEvent.json [hunk 5]
```diff
@@ -1 +1,34 @@
-{"msgParams":{"meta":{"type":"EiffelActivityCanceledEvent","version":"1.1.0","tags":["my_meta[0]tags","my_meta[1]tags"],"source":{"domainId":"my_meta.source.domainId","host":"my_meta.source.host","name":"my_meta.source.name","uri":"my_meta.source.uri"},"security":{"sdm":{"authorIdentity":"my_meta.security.sdm.authorIdentity","encryptedDigest":"my_meta.security.sdm.encryptedDigest"}}}},"eventParams":{"data":{"reason":"my_data.reason","customData":[{"key":"my.data.customData[0]key","value":"my.data.customData[0]value"},{"key":"my.data.customData[1]key","value":"my.data.customData[1]value"}]},"links":[{"type":"ACTIVITY_EXECUTION","target":"e269b37d-17a1-4a10-aafb-c108735ee51f"},{"type":"CAUSE","target":"e269b37d-17a1-4a10-aafb-c108735ee50a"},{"type":"FLOW_CONTEXT","target":"e269b37d-17a1-4a10-aafb-c108735ee48a"}]}}
+{
+	"msgParams": {
+		"meta": {
+			"type": "EiffelActivityCanceledEvent",
+			"version": "3.0.0",
+			"tags": ["my_meta[0]tags", "my_meta[1]tags"],
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityCanceledEvent.json]: correct the tags. Follow one pattern for meta in all eventas.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityCanceledEvent.json [hunk 6]
```diff
@@ -1 +1,34 @@
...
+				"domainId": "my_meta.source.domainId",
+				"host": "my_meta.source.host",
+				"name": "my_meta.source.name",
+				"uri": "my_meta.source.uri"
+			}
+		}
+	},
+	"eventParams": {
+		"data": {
+			"reason": "my_data.reason",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityCanceledEvent.json]: use some valid reason.

File: publish-service/src/test/resources/EiffelActivityFinishedEvent.json [hunk 7]
```diff
@@ -1,37 +1,51 @@
...
-		}
-	}
+  "meta": {
+    "id": "0238a8bd-9bdf-4161-aeff-b00eccf92983",
+    "type": "EiffelActivityFinishedEvent",
+    "version": "3.0.0",
+    "time": 1554385219068,
+    "tags": [
+     "product.development",
+     "producet.feature1"
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/EiffelActivityFinishedEvent.json]: product.

File: publish-service/src/test/resources/EiffelActivityFinishedEvent.json [hunk 8]
```diff
@@ -1,37 +1,51 @@
...
+      {
+        "key": "otherLog",
+        "value": "http://myHost.com/firstLog33"
+      }
+    ]
+  },
+  "links": [
+    {
+      "type": "ACTIVITY_EXECUTION",
+      "target": "aaaaaaaa-bbbb-5ccc-8ddd-eeeeeeeeeee1"
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/EiffelActivityFinishedEvent.json]: invalid ACTIVITY_EXECUTION link.

File: publish-service/src/test/resources/EiffelActivityFinishedEvent.json [hunk 9]
```diff
@@ -1,37 +1,51 @@
...
+      {
+        "name": "firstLog",
+        "uri": "http://myHost.com/firstLog"
+      },
+      {
+        "name": "otherLog",
+        "uri": "http://myHost.com/firstLog33"
+      }
+    ],
+    "customData": [
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/EiffelActivityFinishedEvent.json]: remove customdata or add some meaningful ones.

File: publish-service/src/test/resources/EiffelActivityFinishedEvent.json [hunk 10]
```diff
@@ -1,37 +1,51 @@
...
+    }
+  },
+  "data": {
+    "outcome": {
+      "conclusion": "TIMED_OUT",
+      "description": "Compilation timed out."
+    },
+    "persistentLogs": [
+      {
+        "name": "firstLog",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/EiffelActivityFinishedEvent.json]: correct information in persistentLogs

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityCanceledEvent.json [hunk 11]
```diff
@@ -1 +1,34 @@
...
+				"domainId": "eiffeltest",
+				"host": "localhost",
+				"name": "LOCALHOST",
+				"uri": "https://localhost:8080/jenkins/job"
+			}
+		}
+	},
+	"eventParams": {
+		"data": {
+			"reason": "my_data.reason",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityCanceledEvent.json]: Use valid reason

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityFinishedEvent.json [hunk 12]
```diff
@@ -1 +1,39 @@
...
+		}
+	},
+	"eventParams": {
+		"data": {
+			"outcome": {
+				"conclusion": "SUCCESSFUL",
+				"description": "DONE"
+			},
+			"persistentLogs": [{
+            "name": "firstLog",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityFinishedEvent.json]: format the json

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityStartedEvent.json [hunk 13]
```diff
@@ -1 +1,44 @@
...
+		}
+	},
+	"eventParams": {
+		"data": {
+			"executionUri": "http://localhost:8080/job/dsd",
+			"liveLogs": [{
+            "name": "console",
+            "uri": "http://localhost:8085/job/dsd/9/console"
+          }],
+			"customData": [ {
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityStartedEvent.json]: format the json

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityTriggeredEvent.json [hunk 14]
```diff
@@ -0,0 +1,43 @@
+{
+    "msgParams": {
+        "meta": {
+            "type": "EiffelActivityTriggeredEvent",
+            "version": "3.0.0",
+            "tags": ["product_master","product_feature1"],
+			"source": {
+				"domainId": "eiffeltest",
+				"host": "localhost",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityTriggeredEvent.json]: format the json, All Events.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityTriggeredEvent.json [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+				"uri": "https://localhost:8080/jenkins/job"
+			}
+        }
+    },
+    "eventParams": {
+        "data": {
+            "name": "testrem",
+            "executionType":"AUTOMATED",
+            "categories": [
+            "my_data.categories[0]",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelActivityTriggeredEvent.json]: invalid content. Check complete event

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAlertCeasedEvent.json [hunk 16]
```diff
@@ -1 +1,34 @@
...
+				"domainId": "eiffeltest",
+				"host": "localhost",
+				"name": "LOCALHOST",
+				"uri": "https://localhost:8080/jenkins/job"
+			}
+		}
+	},
+	"eventParams": {
+		"data": {
+			"entity": "my.data.entity",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAlertCeasedEvent.json]: wrong content for entity.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAlertRaisedEvent.json [hunk 17]
```diff
@@ -1 +1,38 @@
...
+				"uri": "https://localhost:8080/jenkins/job"
+			}
+		}
+	},
+	"eventParams": {
+		"data": {
+			"heading": "my.data.heading",
+			"body": "my.data.body",
+			"uri": "https://localhost:8080/fem",
+			"entity": "my.data.entity",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAlertRaisedEvent.json]: same here too. heading, body, entity.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAnnouncementPublishedEvent.json [hunk 18]
```diff
@@ -1 +1,36 @@
...
+			"customData": [{
+            "key": "EIFFELVERSION",
+            "value": "eiffel2.0"
+          }]
+		},
+		"links": [{
+			"type": "ARTIFACT",
+			"target": "e269b37d-17a1-4a10-aafb-c108735ee51f"
+		}, {
+			"type": "CAUSE or CONTEXT or FLOW_CONTEXT - optional",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAnnouncementPublishedEvent.json]: wrong content

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAnnouncementPublishedEvent.json [hunk 19]
```diff
@@ -1 +1,36 @@
...
+				"domainId": "eiffeltest",
+				"host": "localhost",
+				"name": "LOCALHOST",
+				"uri": "https://localhost:8080/jenkins/job"
+			}
+		}
+	},
+	"eventParams": {
+		"data": {
+			"locations": [{
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAnnouncementPublishedEvent.json]: how come locations is present in EiffelAnnouncementPublishedEvent?

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAnnouncementPublishedEvent.json [hunk 20]
```diff
@@ -1 +1,36 @@
...
+                "type": "ARTIFACTORY",
+                "uri": "https://localhost:8080/nexus/"
+            }]
+			"customData": [{
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelAnnouncementPublishedEvent.json]: wrong json structure also. ', 'expected before customData

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelArtifactCreatedEvent.json [hunk 21]
```diff
@@ -1 +1,51 @@
...
+{
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelArtifactCreatedEvent.json]: File is invalid. when i try to format i get these errors.
Error:Invalid character found at position 34.[Code 180, Structure 63]
Error:Expecting comma or ], not string.[Code 10, Structure 64]
Error:Strings should be wrapped in double quotes.[Code 17, Structure 64]
Error:Expecting comma or ], not colon.[Code 10, Structure 65]
Error:Expecting comma or ], not colon.[Code 10, Structure 69]
Error:Expecting comma or ], not colon.[Code 10, Structure 73]
Error:Expecting comma or ], not colon.[Code 10, Structure 77]
Error:Expecting comma or ], not string.[Code 10, Structure 80]
Error:Strings should be wrapped in double quotes.[Code 17, Structure 80]
Error:Expecting comma or ], not colon.[Code 10, Structure 81]
Error:Strings should be wrapped in double quotes.[Code 17, Structure 82]
Error:Expecting comma or ], not colon.[Code 10, Structure 86]
Error:Expecting comma or ], not colon.[Code 10, Structure 92]
Error:Expecting comma or ], not }.[Code 6, Structure 114]
Error:Expecting comma or ], not }.[Code 6, Structure 170]
Error:Expecting closing } at end[Code 22, Structure 170]
Error:Expecting closing } at end[Code 22, Structure 170]
Error:Expecting closing } at end[Code 22, Structure 170]

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelArtifactCreatedEvent.json [hunk 22]
```diff
@@ -1 +1,51 @@
...
+        "name": "LOCALHOST",
+        "uri": "https://localhost:8080/jenkins/job"
+        }
+    }
+    },
+  "eventParams": {
+    "data": {
+        "fileInformation": [{
+        "name": "CMPAC",
+                "tags": ["my_data.fileInformation[0].tags]
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelArtifactCreatedEvent.json]: invalid information. Correct and check in entire event.,

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelConfidenceLevelModifiedEvent.json [hunk 23]
```diff
@@ -1 +1,59 @@
...
+  },
+  "eventParams": {
+    "data": {
+      "name": "ERC",
+      "value": "SUCCESS",
+      "issuer": {
+        "name": "my_data.issuer.name",
+        "email": "my_data.issuer.email",
+        "id": "my_data.issuer.id",
+        "group": "my_data.issuer.group"
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelConfidenceLevelModifiedEvent.json]: invalid information

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelConfidenceLevelModifiedEvent.json [hunk 24]
```diff
@@ -1 +1,59 @@
...
+      },
+      "customData": [
+        {
+          "key": "EIFFELVERSION",
+          "value": "eiffel2.0"
+        },
+        {
+          "key": "COMPNAME",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelConfidenceLevelModifiedEvent.json]: correct this.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelEnvironmentDefinedEvent.json [hunk 25]
```diff
@@ -1 +1,50 @@
...
+    }
+  },
+  "eventParams": {
+    "data": {
+      "name": "ERC",
+      "version": "2.0.0",
+      "image": "486464",
+      "customData": [
+        {
+          "key": "COMPNAME",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelEnvironmentDefinedEvent.json]: correct this.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelIssueDefined.json [hunk 26]
```diff
@@ -0,0 +1,39 @@
...
+      "customData": [
+        {
+          "key": "EIFFELVERSON",
+          "value": "eiffel2.0"
+        }
+      ]
+    },
+    "links": [
+      {
+        "type": "CAUSE or CONTEXT or FLOW_CONTEXT - optional",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelIssueDefined.json]: correct this.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelIssueDefined.json [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+        "target": "required - UUID if type is present"
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelIssueDefined.json]: correct this.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelSourceChangeCreatedEvent.json [hunk 28]
```diff
@@ -1 +1,70 @@
...
+        "name": "LOCALHOST",
+        "uri": "https://localhost:8080/jenkins/job"
+      }
+    }
+  },
+  "eventParams": {
+    "data": {
+      "author": {
+        "name": "fem",
+        "email": "adfs",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelSourceChangeCreatedEvent.json]: this does not look like a valid email.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelSourceChangeCreatedEvent.json [hunk 29]
```diff
@@ -1 +1,70 @@
...
+        "repoName": "testremtem",
+        "repoUri": "https://localhost:8080/fem"
+      },
+      "customData": [
+        {
+          "key": "EIFFELVERSION",
+          "value": "eiffel2.0"
+        },
+        {
+          "key": "COMPNAME",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelSourceChangeCreatedEvent.json]: remove.

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelSourceChangeSubmittedEvent.json [hunk 30]
```diff
@@ -1 +1,62 @@
...
+        "branch": "fem456",
+        "repoName": "eiffeltest"
+      },
+      "customData": [
+        {
+          "key": "EIFFELVERSION",
+          "value": "eiffel2.0"
+        },
+        {
+          "key": "COMPNAME",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelSourceChangeSubmittedEvent.json]: remove

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelTestCaseFinishedEvent.json [hunk 31]
```diff
@@ -1 +1,71 @@
...
+      }
+    }
+  },
+  "eventParams": {
+    "data": {
+      "outcome": {
+        "verdict": "PASSED",
+        "conclusion": "SUCCESSFUL",
+        "description": "SUCCESSFUL",
+        "metrics": [
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelTestCaseFinishedEvent.json]: metrics are like
total 12
paased 11
failed 1

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelTestCaseFinishedEvent.json [hunk 32]
```diff
@@ -1 +1,71 @@
...
+          },
+          {
+            "name": "eiffel",
+            "value": "rem"
+          }
+        ]
+      },
+      "persistentLogs": [
+        {
+          "name": "localhost",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelTestCaseFinishedEvent.json]: valid name

File: publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelTestCaseFinishedEvent.json [hunk 33]
```diff
@@ -1 +1,71 @@
...
+          "uri": "https://localhost:8080/fem"
+        },
+        {
+          "name": "fem",
```
Comment (xvinosi-github) [inline publish-service/src/test/resources/expectedParsedEvents/expected_parsed_EiffelTestCaseFinishedEvent.json]: valid name.

File: publish-service/src/test/resources/testDataForParsingEvents/test_data_for_parsing_EiffelSourceChangeCreatedEvent.json [hunk 34]
```diff
@@ -1,39 +1,29 @@
...
-  "data.author.email":"my_data.author.email",
-  "data.author.id":"my_data.author.id",
-  "data.author.group":"my_data.author.group",
+  "meta.tags[0]":"product_development",
+  "meta.tags[1]":"product_staging",
+  "meta.source.domainId":"eiffeltest",
+  "meta.source.host":"localhost",
+  "meta.source.name":"LOCALHOST",
+  "meta.source.uri":"https://localhost:8080/jenkins/job",
+  "data.author."name": "fem",
```
Comment (SantoshNC68) [inline publish-service/src/test/resources/testDataForParsingEvents/test_data_for_parsing_EiffelSourceChangeCreatedEvent.json]: check the format here?

File: CHANGELOG.md [hunk 35]
```diff
@@ -1,3 +1,6 @@
+## 2.0.0
+- Upgraded eiffel-remrem-semantics version from 1.0.1 to 2.0.3
```
Comment (xvinosi-github) [inline CHANGELOG.md]: You can add something more to the changelog.

**Pull request merged**